modules/storages/app/services/storages/project_storages/bulk_create_service.rb

```diff
@@ -61,14 +60,15 @@ def validate_permissions
       end
     end
 
-    def validate_contract(service_call, project_ids)
+    def validate_contract(service_call, project_ids, params)
+      project_folder_params = params.slice(:project_folder_mode, :project_folder_id)
+
       set_attributes_results = project_ids.map do |id|
-        set_attributes(project_id: id, storage_id: @storage.id, project_folder_mode: @project_folder_mode)
+        set_attributes(project_id: id, storage_id: @storage.id, **project_folder_params)
       end
 
       if (failures = set_attributes_results.select(&:failure?)).any?
-        service_call.success = false
-        service_call.errors = failures.map(&:errors)
+        service_call = failures.first
```

Comment:
Good eye! Yes it is, since the contract is identical in all cases. The main reason we map through all instead of one, is so that we can collect `set_attributes_results` reliably and safely. But in case of contract failure, the errors would be identical since we're just invoking the same context- and we don't have special rules for projects yet.

It does seem a bit odd, so maybe there's a better way to structure it whereby we flatten the errors objects or add a note.